File: src/main/java/io/junseok/mealmate/domain/member/dto/request/SignUpDto.java
package io.junseok.mealmate.domain.member.dto.request;

import io.junseok.mealmate.domain.membercategory.dto.request.CategoryRegister;
import java.util.List;

public record SignUpDto(
    String email,
    String password,
    List<CategoryRegister> categoryRegisters
) {

}

File: src/main/java/io/junseok/mealmate/domain/member/dto/request/SignUpDto.kt
package io.junseok.mealmate.domain.member.dto.request;

import io.junseok.mealmate.domain.membercategory.dto.request.CategoryRegister
import io.junseok.mealmate.domain.membercategory.dto.request.CategoryRegisters
import javax.persistence.Column

data class SignUpDto(
    val email: String,
    val password: String,
    val nickname: String,
    val school: String,
    val department: String,
    val categoryRegisters: List<MemberCategoryRequest>
)

data class MemberCategoryRequest(val categoryName: String)


fun List<MemberCategoryRequest>.toCategoryRegisters() = CategoryRegisters(
    this.map {
        CategoryRegister(
            categoryName = it.categoryName
        )
    }
)
